### PR TITLE
Chat Folders: Shareable folders & Redesign (#502)

### LINKED ISSUE #502: Chat folders: redesign, sharable folders support, bugfixes
Most of chat folders-related features are merged into `main` branch as of commit 286e7a7 (thanks to the PR #297 by @nikita-toropov).

However, there are many TODOs remaining to make it possible to roll out for all users (this list might be incomplete, but will be kept updated):

1. Rework edit screen to match the mockups by @dotvhs
2. When leaving edit screen with unsaved changes, there should be a prompt (same as when editing chat description or other settings and trying to exit the screen)
3. In some cases, the chats list doesn't load fully (e.g. the folder includes 2 chats, but it displays only one. After tapping `Remove from Folder`, the removed chat properly disappears from folder, but the 2nd chat (which should have been already displayed in the list) appears in its place)
4. Sharable folders: support for opening and creating `t.me/addlist` links, short-polling as per TDLib docs,  proper handling on edit screen
4.1. When deleting a shared folder, there should be a warning that all links will be revoked
4.2. When deleting a shared folder, `GetChatFolderChatsToLeave` method should be used, and a pop-up with all chats from the deleting folder should be displayed, with only chats returned from `GetChatFolderChatsToLeave` method selected.
4.3. When shared folder tab is attached, client must call `GetChatFolderNewChats` once in `tdlib.chatFolderUpdatePeriod()` (or immediately after attach event occurs, if it's the first request, or more time elapsed since the previous request), display a preview on top of the screen. Once pressed, the pop-up suggesting to join the new chats should display, which should lead to the `ProcessChatFolderNewChats` call (all of this is covered by mockups in Figma). When none chats are selected, `ProcessChatFolderNewChats` with empty `addedChatIds` should be called to dismiss the chats previously returned in `GetChatFolderChatsToLeave`.
5. Animated version of `Icon + text`: text is shown only one the folder / tab gets selected
6. Some other changes visible in the mockups.
7. After changes above, make it accessible for all users.

<details>
<summary>Mockups and animations by @dotvhs</summary>

**Note:** unread counters in folders are now outlined in `main`, but are outdated in the mockups below. The outlined counters from `main` should be preserved.

* [Figma link](https://www.figma.com/file/LtffmBQ2mze2kGiEoUfm3Y/TGX---Folders)

https://github.com/TGX-Android/Telegram-X/assets/6242627/f9475dba-8253-4438-a348-90600be0e6da
</details>

### PR DESCRIPTION
**Pull request by nikita-toropov**
Fixes #502

### CODE REVIEW

**Review by vkryl — APPROVED** [review]

**Review by Jean-sys77 — COMMENTED** [review]
- [ ]

File: app/src/main/java/org/thunderdog/challegram/MainActivity.java [hunk 1]
```diff
@@ -1245,6 +1246,8 @@ private static ViewController<?> restoreController (BaseActivity context, Tdlib
       restore = new SettingsFoldersController(context, tdlib);
     } else if (id == R.id.controller_editChatFolders) {
       restore = new EditChatFolderController(context, tdlib);
+    } else if (id == R.id.controller_editChatFolderInviteLink) {
```
Comment (stiliniinomj) [inline app/src/main/java/org/thunderdog/challegram/MainActivity.java:1249]: @plusmsgrchatru

**Comment by YepeJ** [comment]
Hola

Enviado desde mi Samsung Mobile de Telcel
Obtener Outlook para Android<https://aka.ms/AAb9ysg>
________________________________

**Review by vkryl — APPROVED** [review]
There are few unfinished things (included/excluded chats screen design, label on active mode, etc), other than that impressive and reliable work. Merging in current state in order to include it while building other PRs.

**Comment by SimoLRepo** [comment]
When do you plan to add this feature?

**Comment by SimoLRepo** [comment]
> go to settings, scroll to bottom, tap version number then experimental settings

I didn't know that, thank you very much.

**Comment by cub561** [comment]
Hi

**Pull request merged**